css/plugins.js

```diff
@@ -1,11 +1,14 @@
-const ExtractTextPlugin = require('extract-text-webpack-plugin');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const IgnoreEmitPlugin = require('ignore-emit-webpack-plugin');
 
 /**
  * Exports the settings for plugins in webpack.config
+ * @param {string} filename determines the name of each output CSS file.
+ * See https://webpack.js.org/plugins/mini-css-extract-plugin/#filename
  */
-module.exports = () => ([
-  new ExtractTextPlugin({
-    filename: 'styles/[name].css',
-    allChunks: true,
+module.exports = (filename) => ([
+  new MiniCssExtractPlugin({
+    filename,
   }),
+  new IgnoreEmitPlugin(/\.js$/),
```

Comment:
`MiniCssExtractPlugin` is the modern replacement for `ExtractTextPlugin`.
`IgnoreEmitPlugin` is necessary to stop `MiniCssExtractPlugin` from trying to emit a js file for each css file.

Also giving `filename` as an argument means we can actually configure this per module. There was an illusion this was possible before, because the modules all included `filename: 'styles/[name].css'` in the `output` config - but that didn't actually work and will actually cause errors now.